src/views/components/textmanagement/TranslateList.tsx

```diff
@@ -61,9 +61,9 @@ const RenderText = ({ text, language }: RenderTextProps) => {
   const { t, i18n } = useTranslation('text_management');
   const languageContext = useContext(LanguageContext);
 
-  const languageToDisplay = i18n.exists(`text_management:language.${language.value.toLowerCase()}`)
-    ? t(`language.${language.value.toLowerCase()}` as never)
-    : t(`language.default`, { prefix: language.value });
+  const languageToDisplay = i18n.exists(`text_management:language_${language.value.toLowerCase()}`)
+    ? t(`language_${language.value.toLowerCase()}` as never)
```

Comment:
Utiliser `as never` est une mauvaise pratique (je l'ai fait par le passé et c'était pas bien ^^').

Si tu veux corriger ce problème de cast il serait préférable d'ajouter une "type guard clause":
```ts

const isLanguageNameInI18n = (value: string): value is 'en' | 'fr' | 'es' | 'it' | 'ko' | 'kana' | 'zh' | 'de' => i18n.exists(`text_management:language_${value}`);

const languageCode = language.value.toLowerCase();
const languageToDisplay = isLanguageNameInI18n(languageCode) ? : i18n.exists(`text_management:${languageCode}`) : t(`language_default`, { prefix: language.value }):
```

Si tout se passe bien, tu devrais pas avoir de problème de typescript parce que la fonction `isLanguageNameInI18n` dit explicitement à typescript quels sont les valeurs possible pour value (donc languageCode) :) 

Par ailleurs, il serait idéal de ne pas avoir à faire `'en' | 'fr' | 'es' | 'it' | 'ko' | 'kana' | 'zh' | 'de'` mais directement récupérer l'info des types de i18n (j'imagine que la config permet d'avoir ça vu qu'on la défini, ça permettra à l'avenir de détecter les erreurs de traduction où un language manque à l'appel).

Je peux malheureusement pas t'aider là dessus maintenant (mon week-end est bouffé par des obligations je peux pas coder :<).

(Cela étant dit, ton code est bon dans le monde de javascript 👍 )

Reply:
Merci pour les conseils qui me permettront de mieux m'améliorer et seront toujours les bienvenus. Pour corriger le problème de `as never` j'ai plutôt opter pour une fonction externe, comme suggérer plus bas, qui factorise la logique de récupération de la langue et ensuite appelée dans les composants `TranlateTarget.tsx` et `TranslateListe.tsx`. Neamoins, je n'ai pas su comment typer les  fonctions `t` et i18n puisque `const { t, i18n } = useTranslation('text_management');` n'est utilisable que dans un composant react et je ne voyait l'utilité d'en créer une dans le cadre d'une factorisation de code. Donc elles sont définies à `any`. Je suis preneuse s'il y a une meilleure façon de le faire :)

Reply:
La factorisation semble être bonne et typescript a pas l'air mécontent.

Pour les types qu'il te manque, tu peux les importer depuis i18next. Le fichier `src/utils/getLanguageDisplayText.ts` deviendrait:

```typescript
import type { TFunction, i18n } from 'i18next';

export const getLanguageDisplayText = (languageKey: string, t: TFunction<'text_management'>, i18n: i18n, prefix?: string) => {
  const languageTextKey = `text_management:language_${languageKey.toLowerCase()}`;
  if (i18n.exists(languageTextKey)) {
    return t(languageTextKey);
  } else {
    return t('language_default', { prefix: prefix ? prefix : languageKey });
  }
};
```

(J'ai juste ajouté l'import type et remplacé les any).

Je te laisse commit ce changement et dans le même temps j'approuve ta PR :)